src/components/accounts/ledger/SetupLedger.js

```diff
@@ -5,28 +5,41 @@ import InstructionsModal from './InstructionsModal';
 import LedgerIcon from '../../svg/LedgerIcon';
 import FormButton from '../../common/FormButton';
 import { Translate } from 'react-localize-redux';
-import { addLedgerAccessKey } from '../../../actions/account'
+import { addLedgerAccessKey, createNewAccount, refreshAccount, removeNonLedgerAccessKeys } from '../../../actions/account'
 import GlobalAlert from '../../responsive/GlobalAlert'
 
 const SetupLedger = (props) => {
 
     const [showInstructions, setShowInstructions] = useState(false);
     const [connect, setConnect] = useState(false);
     const toggleShowInstructions = () => setShowInstructions(!showInstructions);
+    const loading = props.actionsPending.some(action => ['CREATE_NEW_ACCOUNT', 'ADD_LEDGER_ACCESS_KEY', 'REFRESH_ACCOUNT', 'REMOVE_NON_LEDGER_ACCESS_KEYS'].includes(action))
+
+    const handleClick = async () => {
+        setConnect('')
 
-    const onClick = async () => {
-        setConnect('true');
         try {
+            if (props.isNew) {
+                await props.createNewAccount(props.accountId, props.fundingContract, props.fundingKey)
+            }
             await props.addLedgerAccessKey(props.accountId)
-        } catch (e) {
-            return setConnect('fail');
+            await props.refreshAccount()
+        } catch(e) {
+            setConnect('fail');
+            throw(e);
```

Comment:
Why is it needed to throw now? Is this a fix for smth?

nit: also can be just `throw e;`, it's not a function call